scripts/core/superdesk-api.d.ts

```diff
@@ -2154,6 +2155,7 @@ declare module 'superdesk-api' {
     export interface IPreviewComponentProps {
         item: IArticle;
         value: any;
+        field: IVocabulary;
```

Comment:
It'd be cleaner to only expose field_id. On the other hand, you're only using it to retrieve a label, which shouldn't be done inside the extenion so maybe it's not needed to expose it at all.

Reply:
Please see my comment in the `superdesk-tga` repo regarding the labels.

Reply:
In regards to the `field` param, I agree it may be cleaner to only expose the bare minimum that **current** extensions are using, but I do not see an issue with passing the entire `IVocabulary` to the extension considering that data belongs to said extension and is not outside of it's scope anyway.

Reply:
When designing the API I'm thinking about many use cases, not only about current ones, but I can't think of why would an input field need a vocabulary. Custom fields by design are not supposed to render labels themselves, and for field configs, we have dedicated config interface and component. Also not all fields may have an associated vocabulary - for example for rundown items(that are built on top of refactored authoring) I'm using inline JSON as a content profile and there are no associated vocabularies in that case.

I think an easy and clean solution here, if one really needed a vocabulary, would be to pass field ID and fetch the vocabulary inside field editor/preview component.